nodeJS/week3/app.js

```diff
@@ -0,0 +1,80 @@
+import knex from 'knex'
+const knexInstance = knex({
+  client: "mysql2",
+  connection: {
+    host: process.env.DB_HOST || "127.0.0.1",
+    port: process.env.DB_PORT || 3306,
+    user: process.env.DB_USER || "root",
+    password: process.env.DB_PASSWORD || "my-secret-pw",
+    database: process.env.DB_NAME || "hyf_node_week3_warmup",
+    multipleStatements: true,
+  },
+});
+
+import express from 'express'
+const app = express();
+const port = process.env.PORT || 3000;
+
+app.use(express.json());
+
+const apiRouter = express.Router();
+app.use("/api", apiRouter);
+
+const contactsAPIRouter = express.Router();
+apiRouter.use("/contacts", contactsAPIRouter);
+
+/* contactsAPIRouter.get("/", async (req, res) => {
+  let query = knexInstance.select("*").from("contacts");
+
+  if ("sort" in req.query) {
+    const orderBy = req.query.sort.toString();
+    if (orderBy.length > 0) {
+      query = query.orderByRaw(orderBy);
+    }
+  }
+
+  console.log("SQL", query.toSQL().sql);
+
+  try {
+    const data = await query;
+    res.json({ data });
+  } catch (e) {
+    console.error(e);
+    res.status(500).json({ error: "Internal server error" });
+  }
+}); */
+
+contactsAPIRouter.get("/", async (req, res) => {
+    let query = knexInstance.select("*").from("contacts");
+  
+    if ("sort" in req.query) {
+      const orderBy = req.query.sort.toString().split(' ');
+      const column = orderBy[0];
+      const direction = orderBy[1] ? orderBy[1].toLowerCase() : 'asc';
+  
+      const validSortColumns = ['id', 'first_name', 'last_name', 'email', 'phone'];
+      const validSortDirections = ['asc', 'desc'];
+  
+      // Validate sort parameters
+      if (validSortColumns.includes(column) && validSortDirections.includes(direction)) {
+        query = query.orderBy(column, direction);
+      } else {
+        return res.status(400).json({ error: "Invalid sort parameter" });
+      }
+    }
+  
+    console.log("SQL", query.toSQL().sql);
+  
+    try {
+      const data = await query;
+      res.json({ data });
+    } catch (e) {
+      console.error(e);
+      res.status(500).json({ error: "Internal server error" });
```

Comment:
no need to write Internal server error here, it is already communicated in the 500 code.